File: client/src/components/AccessibilityMenu.js
import React, { useEffect, useState } from "react";
import "./AccessibilityMenu.css";

export default function AccessibilityMenu() {
  const [open, setOpen] = useState(false);
  const [fontSize, setFontSize] = useState(100);
  const [highContrast, setHighContrast] = useState(false);
  const [grayscale, setGrayscale] = useState(false);
  const [underlineLinks, setUnderlineLinks] = useState(false);
  const [readableFont, setReadableFont] = useState(false);

  useEffect(() => {
    document.documentElement.style.fontSize = `${fontSize}%`;
    document.body.classList.toggle("accessibility-high-contrast", highContrast);
    document.body.classList.toggle("accessibility-grayscale", grayscale);
    document.body.classList.toggle("accessibility-underline-links", underlineLinks);
    document.body.classList.toggle("accessibility-readable-font", readableFont);
  }, [fontSize, highContrast, grayscale, underlineLinks, readableFont]);

  const resetAccessibility = () => {
    setFontSize(100);
    setHighContrast(false);
    setGrayscale(false);
    setUnderlineLinks(false);
    setReadableFont(false);
  };

  return (
    <div className="accessibility-wrapper" dir="rtl">
      <button
        type="button"
        className="accessibility-floating-button"
        onClick={() => setOpen((current) => !current)}
        aria-expanded={open}
        aria-controls="accessibility-panel"
        aria-label="פתיחת תפריט נגישות"
      >
        ♿
      </button>

      {open && (
        <section id="accessibility-panel" className="accessibility-panel">
          <div className="accessibility-panel__header">
            <h2>תפריט נגישות</h2>
            <button
              type="button"
              className="accessibility-close"
              onClick={() => setOpen(false)}
              aria-label="סגירת תפריט נגישות"
            >
              ✕
            </button>
          </div>

          <div className="accessibility-font-controls">
            <button type="button" onClick={() => setFontSize((size) => Math.min(size + 10, 150))}>
              הגדלת טקסט
            </button>
            <button type="button" onClick={() => setFontSize((size) => Math.max(size - 10, 80))}>
              הקטנת טקסט
            </button>
          </div>

          <button type="button" className={highContrast ? "active" : ""} onClick={() => setHighContrast((value) => !value)}>
            ניגודיות גבוהה
          </button>

          <button type="button" className={grayscale ? "active" : ""} onClick={() => setGrayscale((value) => !value)}>
            גווני אפור
          </button>

          <button type="button" className={underlineLinks ? "active" : ""} onClick={() => setUnderlineLinks((value) => !value)}>
            הדגשת קישורים
          </button>

          <button type="button" className={readableFont ? "active" : ""} onClick={() => setReadableFont((value) => !value)}>
            גופן קריא
          </button>

          <button type="button" className="accessibility-reset" onClick={resetAccessibility}>
            איפוס נגישות
          </button>
        </section>
      )}
    </div>
  );
}

File: client/src/components/AccessibilityMenu.cs
/* AccessibilityMenu.css - FULL REPLACE */

.accessibility-wrapper {
  position: fixed;
  left: 18px;
  bottom: 18px;
  z-index: 9999;
}

.accessibility-floating-button {
  width: 64px;
  height: 64px;
  border: 3px solid #ffffff;
  border-radius: 50%;
  background: linear-gradient(180deg, #247cf2, #0a48ae);
  color: #ffffff;
  font-size: 34px;
  box-shadow: 0 10px 24px rgba(0, 0, 0, 0.38);
  cursor: pointer;
}

.accessibility-panel {
  position: absolute;
  left: 0;
  bottom: 78px;
  width: min(340px, calc(100vw - 36px));
  padding: 18px;
  border: 2px solid #75b5ff;
  border-radius: 18px;
  background: #082558;
  color: #ffffff;
  box-shadow: 0 18px 40px rgba(0, 0, 0, 0.45);
}

.accessibility-panel__header {
  display: flex;
  align-items: center;
  justify-content: space-between;
  gap: 12px;
  margin-bottom: 16px;
}

.accessibility-panel__header h2 {
  margin: 0;
}

.accessibility-panel button {
  width: 100%;
  min-height: 48px;
  margin-top: 10px;
  padding: 10px 14px;
  border: 2px solid #7fb7ff;
  border-radius: 12px;
  background: #ffffff;
  color: #093f91;
  font-weight: 900;
  cursor: pointer;
}

.accessibility-panel button:hover,
.accessibility-panel button.active {
  background: #ffd500;
  color: #071a3c;
}

.accessibility-font-controls {
  display: grid;
  grid-template-columns: 1fr 1fr;
  gap: 10px;
}

.accessibility-font-controls button {
  margin-top: 0;
}

.accessibility-panel .accessibility-reset {
  background: #d52c42;
  color: #ffffff;
  border-color: #ff8a99;
}

.accessibility-high-contrast {
  background: #000000 !important;
  color: #ffff00 !important;
}

.accessibility-high-contrast * {
  background-color: #000000 !important;
  color: #ffff00 !important;
  border-color: #ffff00 !important;
  box-shadow: none !important;
}

.accessibility-grayscale {
  filter: grayscale(1);
}

.accessibility-underline-links a {
  text-decoration: underline !important;
  text-decoration-thickness: 3px !important;
  text-underline-offset: 4px !important;
}

.accessibility-readable-font,
.accessibility-readable-font * {
  font-family: Arial, Verdana, sans-serif !important;
  letter-spacing: 0.03em;
  line-height: 1.65;
}

@media (max-width: 600px) {
  .accessibility-wrapper {
    left: 10px;
    bottom: 10px;
  }

  .accessibility-floating-button {
    width: 56px;
    height: 56px;
    font-size: 30px;
  }
}
